src/query/interpret/eval.hpp

```diff
@@ -175,13 +175,15 @@ class PrimitiveLiteralExpressionEvaluator : public ExpressionVisitor<TypedValue>
 class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
  public:
   ExpressionEvaluator(Frame *frame, const SymbolTable &symbol_table, const EvaluationContext &ctx, DbAccessor *dba,
-                      storage::View view, FrameChangeCollector *frame_change_collector = nullptr)
+                      storage::View view, FrameChangeCollector *frame_change_collector = nullptr,
+                      bool treat_null_as_zero = false)
       : frame_(frame),
         symbol_table_(&symbol_table),
         ctx_(&ctx),
         dba_(dba),
         view_(view),
-        frame_change_collector_(frame_change_collector) {}
+        frame_change_collector_(frame_change_collector),
+        treat_null_as_zero_(treat_null_as_zero) {}
```

Comment:
It's a clutch. The reason for doing this is following.
If weight function (it calculates weight) uses node property inside, it should include both start and end vertices. For example, you can have 3 nodes and 2 edges. For one of input node input edge for weight function will be null. But we expect to have Numeric (int or double) or Duration type of expression. When one of operand is Null the whole expression is Null. For example,
`MATCH path=(:station {name:"A"})-[*WSHORTEST (r, v | v.departure - v.arrival + r.duration) total_weight (r,n,p,w | (nodes(p)[-1]).name > (nodes(p)[-2]).name AND not(w is null))]->(:station {name:"C"}) RETURN path, total_weight;`
In this example I use both node and edge properties in weight lambda. If r is null, then whole expression `v.departure - v.arrival + r.duration` will be null, which is not expected.

Reply:
Could this be solved by using the `coalesce`?

Reply:
I checked this example and yes, it could be solved by `coalesce`.
It can be rewritten as:
```
MATCH path=(:station {name:"A"})-[*WSHORTEST (r, v | v.departure - v.arrival + coalesce(r.duration, 0)) total_weight (r,n,p,w | (nodes(p)[-1]).name > (nodes(p)[-2]).name AND not(w is null))]->(:station {name:"C"}) RETURN path, total_weight;
```
Question: do we want to user write like this? I think that's possible and we can describe this in the documentation.

Reply:
Updated the PR. Now we don't have `treat_null_as_zero`.